akka-http-core/src/main/scala/akka/http/impl/model/parser/CommonActions.scala

```diff
@@ -32,24 +31,27 @@ private[parser] trait CommonActions {
         case custom        ⇒ MediaType.customMultipart(custom, params)
       }
       case mainLower ⇒
-        if (areNoCustomMediaTypesDefined) // try to prevent closure creation for usual case without custom media types
```

Comment:
Rather than having this condition I have simply inlined `Option.fold`. This allows the `areNoCustomMediaTypesDefined` method to be removed and the`getPredefinedOrCreateCustom` method can be inlined and removed.

To me this reads better, but I could revert this if desired.

Reply:
Doesn't this change mean `customMediaTypes(mainLower, subLower)` will now be called even when there are no custom media types defined?